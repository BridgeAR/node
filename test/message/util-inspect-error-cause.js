'use strict';

require('../common');

class FoobarError extends Error {
  status = 'Feeling good';
}

const cause1 = new TypeError('Inner error');
const cause2 = new FoobarError('Individual message', { cause: cause1 });
cause2.extraProperties = 'Yes!';
const cause3 = new Error('Stack causes', { cause: cause2 });

process.nextTick(() => {
  const error = new RangeError('New Stack Frames', { cause: cause2 });
  const error2 = new RangeError('New Stack Frames', { cause: cause3 });

  console.log(error);
  console.log(cause3);
  console.log(error2);
});
