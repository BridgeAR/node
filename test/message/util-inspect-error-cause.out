RangeError: New Stack Frames
    at *
    at * {
  [cause]: FoobarError: Individual message
      at *
      at *
      ... 4 lines matching cause stack trace ...
      at * {
    status: 'Feeling good',
    extraProperties: 'Yes!',
    [cause]: TypeError: Inner error
        at *
        at *
        at *
        at *
        at *
        at *
        at *
  }
}
Error: Stack causes
    at *
    at *
    ... 4 lines matching cause stack trace ...
    at * {
  [cause]: FoobarError: Individual message
      at *
      at *
      ... 4 lines matching cause stack trace ...
      at *
    status: 'Feeling good',
    extraProperties: 'Yes!',
    [cause]: TypeError: Inner error
        at *
        at *
        at *
        at *
        at *
        at *
        at *
  }
}
RangeError: New Stack Frames
    at *
    at * {
  [cause]: Error: Stack causes
      at *
      at *
      ... 4 lines matching cause stack trace ...
      at * {
    [cause]: FoobarError: Individual message
        at *
        at *
        ... 4 lines matching cause stack trace ...
        at * {
      status: 'Feeling good',
      extraProperties: 'Yes!',
      [cause]: TypeError: Inner error
          at *
          at *
          at *
          at *
          at *
          at *
          at *
    }
  }
}
